src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -111,52 +109,39 @@ public void register(Router router) {
       .handler(handle(this::markAsUnknown));
     router.post(RELATIVE_ITEMS_PATH_ID + "/mark-withdrawn")
       .handler(handle(this::markAsWithdrawn));
-
   }
 
-  private CompletableFuture<Void> markAsWithdrawn(
-    RoutingContext routingContext, WebContext webContext, Clients clients) {
-
-    final MoveItemIntoStatusService moveItemIntoStatusService = new MoveItemIntoStatusService(storage
-      .getItemCollection(webContext), clients);
-
-    return moveItemIntoStatusService.processMarkItemWithdrawn(webContext)
-      .thenAccept(item -> respondWithItemRepresentation(item, HTTP_CREATED.toInt(),
-          routingContext, webContext));
+  private void registerMarkItemAsHandler(String itemStatusUrl, Router router) {
+    router.post(RELATIVE_ITEMS_PATH_ID + itemStatusUrl)
+      .handler(handle(this::markItemAsTargetStatus));
   }
 
-  private CompletableFuture<Void> markAsInProcess(
+  private CompletableFuture<Void> markItemAsTargetStatus(
     RoutingContext routingContext, WebContext webContext, Clients clients) {
+    final var itemStatusName = ItemStatusURL.getItemStatusNameForUrl(routingContext.request().uri());
```

Comment:
When a method signature spans multiple lines, I find it easier to differentiate between the signature and the body of the method if there is whitespace between them. Please introduce a blank line between the signature and the first line.